src/modules/fw_pos_control/FixedwingPositionControl.cpp

```diff
@@ -1236,7 +1240,9 @@ FixedwingPositionControl::control_auto_loiter(const float control_interval, cons
 		// We're in a loiter directly before a landing WP. Enable our landing configuration (flaps,
 		// landing airspeed and potentially tighter altitude control) already such that we don't
 		// have to do this switch (which can cause significant altitude errors) close to the ground.
-		_tecs.set_altitude_error_time_constant(_param_fw_thrtc_sc.get() * _param_fw_t_h_error_tc.get());
+
+		_is_low_height = true; // In low-height flight, TECS will control altitude tighter
```

Comment:
Please don't set this as a global variable, but try to input it as part of the method's parameters (e.g. `control_auto_loiter`) here. Same with other places that this is defined.